src/scripts/components/layers/layer-legend/layer-legend.tsx

```diff
@@ -1,4 +1,4 @@
-import React, {FunctionComponent} from 'react';
+import React, {FunctionComponent, useState} from 'react';
```

Comment:
I think it would be cleaner to use a separate component for this type of legend. Would be less confusing and who knows which other legend types will come in the future